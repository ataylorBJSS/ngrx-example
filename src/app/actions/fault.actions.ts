import { Injectable } from '@angular/core';
import { Action } from '@ngrx/store';
import { IFault } from './../models/fault.model';

export enum actionType {
  ADD_FAULT = '[FAULT] Add',
  REMOVE_FAULT = '[FAULT] Remove'
}

export class AddFault implements Action {
  readonly type = actionType.ADD_FAULT;

  constructor(public payload: IFault) { }
}

export class RemoveFauilt implements Action {
  readonly type = actionType.REMOVE_FAULT;

  constructor(public payload: string) { }
}

export type Actions = AddFault | RemoveFauilt;
