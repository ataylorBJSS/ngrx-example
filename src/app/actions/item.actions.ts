import { Injectable } from '@angular/core';
import { Action } from '@ngrx/store';
import { Item } from './../models/item.model';
import { Update } from "@ngrx/entity";

export enum actionType {
  ADD_ITEM = '[ITEM] Add',
  REMOVE_ITEM = '[ITEM] Remove',
  UPDATE_ITEM = '[ITEM] Update'
}

export class AddItem implements Action {
  readonly type = actionType.ADD_ITEM;

  constructor(public payload: { item: Item }) { }
}

export class UpdateItem implements Action {
  readonly type = actionType.UPDATE_ITEM;

  constructor(public payload: { item: Update<Item> }) { }
}
export class RemoveItem implements Action {
  readonly type = actionType.REMOVE_ITEM;

  constructor(public payload: { id: string }) { }
}

export type Actions = AddItem | UpdateItem | RemoveItem;
