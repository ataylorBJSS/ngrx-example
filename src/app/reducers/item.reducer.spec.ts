import { reducer } from "./item.reducer";
import * as fromItems from "./item.reducer";
import { AddItem, RemoveItem, UpdateItem } from "../actions/item.actions";
import { Item, generateMockItem } from "../models/item.model";

describe('ItemReducer', () => {

  const item1 = generateMockItem();
  const item2 = { ...item1, id: '2' };
  const item3 = { ...item1, id: '3' };

  const initialState: fromItems.State = {
    ids: [item1.id, item2.id],
    entities: {
      [item1.id]: item1,
      [item2.id]: item2,
    }
  };

  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

    });
  });


});
