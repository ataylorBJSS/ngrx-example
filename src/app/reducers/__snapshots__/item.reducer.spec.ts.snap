exports[`ItemsReducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "ids": Array []
}
`;

exports[`BooksReducer LOAD should add a single book, if the book does not exist 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "id": "1",
      "volumeInfo": Object {
        "authors": Array [
          "author",
        ],
        "averageRating": 3,
        "description": "description",
        "imageLinks": Object {
          "smallThumbnail": "string",
          "thumbnail": "string",
        },
        "publishDate": "",
        "publisher": "publisher",
        "ratingsCount": 5,
        "subtitle": "subtitle",
        "title": "title",
      },
    },
  },
  "ids": Array [
    "1",
  ],
  "selectedBookId": null,
}
`;
